### DISCUSSION TITLE
Playback breakpoints

### DISCUSSION THREAD

**Discussion by jtbandes** [foxglove/community#230] [Q&A]
Currently the app provides no way to "find" instances of messages on a certain topic other than by seeking back/forward until the desired message is displayed. If the messages on the desired topic are very high-frequency (such as /tf) it can be very difficult to find and display the desired message in the app.

The app could support a "breakpoints" feature where playback would automatically stop when some kind of event occurs. Such an event could be:
- A certain receiveTime is reached (this could be used to implement forward/back arrow keys as per foxglove/studio#2270)
- A message is received on a certain topic
- A message is received on a certain topic matching a user-defined predicate (e.g. `header.frame_id="parent" AND child_frame_id="child"`)

See also: https://github.com/orgs/foxglove/discussions/885#discussion-5903400

**Comment by sachitv** [comment]
It would be great if it was possible to somehow expand the timeline view with messages on a per topic basis similar to what rerun does:
![full](https://github.com/foxglove/community/assets/34048126/97779b47-2287-49ab-9f3b-13b06a95869b)

Thereafter provide a shortcut keystroke for both:
1. Go to the next time for the selected topic
2. Go to the next time for the log
